tests/31sync/15lazy-members.pl

```diff
@@ -200,6 +200,167 @@
       });
    };
 
+
+test "We don't send redundant membership state across incremental syncs by default",
+   requires => [ local_user_fixtures( 3 ),
+                 qw( can_sync ) ],
+
+   check => sub {
+      my ( $alice, $bob, $charlie ) = @_;
+
+      # Alice creates a public room,
+      # Bob and Charlie join
+      # Bob sends 10 events into it
+      # Charlie sends 5 events into it
+      # Alice syncs with a filter on the last 10 events, and lazy loaded members
+      # Alice should see only Bob and Charlie in the membership list.
+      # Bob sends 1 more event
+      # Charlie sends 1 more event
+      # Alice syncs again; she should not see any membership events as
+      # the redundant ones for Bob and Charlie are removed.
+
+      # TODO: speed up time and check that if we wait an hour then the server's
```

Comment:
isn't this an implementation issue? servers can keep the cache for as long or as short as they want?